src/main/kotlin/ui/PRComponent.kt

```diff
@@ -147,47 +127,83 @@ class OwnPRComponent(ownPR: PR,
                      awtExecutor: Executor)
     : PRComponent(ownPR, imagesSource, awtExecutor) {
 
-    override fun addApproveButton(buttonSize: Dimension) {
+    override fun createApproveButton() {
         //Do not add an approve button for own PRs
     }
 }
 
-class ReviewersPanel() : JPanel(FlowLayout(FlowLayout.LEFT, 2, 2)) {
-    private val approvers = JPanel(FlowLayout(FlowLayout.LEFT, 2, 2))
-    private val reviewers = JPanel(FlowLayout(FlowLayout.LEFT, 2, 2))
-    private val approvedBy = JBLabel("Approved:")
-    private val reviewBy = JBLabel("Reviewers:")
-    private val others = JBLabel("Others:")
+class ReviewersPanel(reviewers: MutableList<PRParticipant>,
+                     imagesSource: MediaSource<BufferedImage>,
+                     awtExecutor: Executor) : JPanel(HorizontalLayout(5)) {
+    companion object {
+        const val ALWAYS_DISPLAY_REVIEWERS_COUNT = 5
+    }
 
     init {
-        approvedBy.isVisible = false
-        others.isVisible = false
-        reviewBy.isVisible = false
-        add(approvedBy)
-        add(approvers)
-        add(others)
-        add(reviewBy)
-        add(reviewers)
+        this.isOpaque = false
+        reviewers.sortWith(Comparator { o1, o2 -> o1.status.compareTo(o2.status) })
+        val labels: Map<PRParticipant, ReviewerItem> = reviewers.associateWith { prParticipant -> ReviewerItem(prParticipant) }
+
+        val alwaysVisibleReviewerCount = Math.min(ALWAYS_DISPLAY_REVIEWERS_COUNT, reviewers.size)
+
+        reviewers.take(alwaysVisibleReviewerCount).forEach { this.add(labels[it]) }
+
+        val reviewersInCombo = reviewers.size - alwaysVisibleReviewerCount
```

Comment:
Minor:
If we have exactly `alwaysVisibleReviewerCount + 1` reviewer we will hide the last one under `+1` button, which doesn't make sense, becuase we can just add avatar instead of `+1`

Reply:
Сompletely agree
Fixed in b9313ef2567